AWSAppSyncClient/Internal/SQLiteSerialization.swift

```diff
@@ -45,7 +45,7 @@ final class SQLiteSerialization {
         switch fieldJSONValue {
         case let dictionary as JSONObject:
             guard let reference = dictionary[serializedReferenceKey] as? String else {
-                throw AWSAppSyncQueriesCacheError.invalidRecordValue(value: fieldJSONValue)
+                return fieldJSONValue
```

Comment:
From the Apollo issue linked in the description, has the PR with the same change to the `SQLiteSerialization` file here https://github.com/apollographql/apollo-ios/pull/1144/files#diff-e9bbcdac7903360aed17a5df82e644121e800adbc8b56bc03ffcf1dccc7dfe0a . With this, this change looks accurate to do since I think the file was lifted from the apollo codebase.